frontend/utils/distribution/group-income-distribution.test.js

```diff
@@ -0,0 +1,186 @@
+/* eslint-env mocha */
+
+import should from 'should'
+import groupIncomeDistribution from './group-income-distribution.js'
+
+const mincomeAmount = 1000
+let setup = [
+  { type: 'startCycleEvent', data: { cycle: 0, latePayments: [] } }
+]
+
+describe('Test group-income-distribution.js', function () {
+  it('Test empty distirbution event list for un-adjusted distribution.', function () {
+    should(groupIncomeDistribution(setup, { adjusted: false, minimizeTxns: false, mincomeAmount })).eql([])
+  })
+  it('EVENTS: [u1, u5, u2 and u3] join the group and set haveNeeds of [100, 100, -50, and -50], respectively. Test unadjusted.', function () {
+    setup = [setup,
+      { type: 'haveNeedEvent', data: { name: 'u1', haveNeed: 100 } },
+      { type: 'haveNeedEvent', data: { name: 'u5', haveNeed: 100 } },
+      { type: 'haveNeedEvent', data: { name: 'u2', haveNeed: -50 } },
+      { type: 'haveNeedEvent', data: { name: 'u3', haveNeed: -50 } }
+    ].flat()
```

Comment:
Recommend adjustments / alternatives to doing it with `.flat()`

- Option 1: `setup = [...setup, { type: 'haveNeedEvent' .. }]`
- Option 2: `setup = setup.concat([{ type: 'haveNeedEvent' .. }, ..])`

Reply:
@taoeffect Again, this is (after) the next item on the TODO list. Going to push cypress fixes, first.

Reply:
Went with Option 1. You'll see it in the next push.